p11-kit/proxy.c

```diff
@@ -1743,36 +2164,54 @@ static CK_X_FUNCTION_LIST proxy_functions = {
 	proxy_C_SeedRandom,
 	proxy_C_GenerateRandom,
 	proxy_C_WaitForSlotEvent,
+	/* PKCS #11 3.0 */
+	proxy_C_LoginUser,
+	proxy_C_SessionCancel,
+	proxy_C_MessageEncryptInit,
+	proxy_C_EncryptMessage,
+	proxy_C_EncryptMessageBegin,
+	proxy_C_EncryptMessageNext,
+	proxy_C_MessageEncryptFinal,
+	proxy_C_MessageDecryptInit,
+	proxy_C_DecryptMessage,
+	proxy_C_DecryptMessageBegin,
+	proxy_C_DecryptMessageNext,
+	proxy_C_MessageDecryptFinal,
+	proxy_C_MessageSignInit,
+	proxy_C_SignMessage,
+	proxy_C_SignMessageBegin,
+	proxy_C_SignMessageNext,
+	proxy_C_MessageSignFinal,
+	proxy_C_MessageVerifyInit,
+	proxy_C_VerifyMessage,
+	proxy_C_VerifyMessageBegin,
+	proxy_C_VerifyMessageNext,
+	proxy_C_MessageVerifyFinal
 };
 
-#ifdef OS_WIN32
-__declspec(dllexport)
-#endif
-CK_RV
-C_GetFunctionList (CK_FUNCTION_LIST_PTR_PTR list)
+static int
+internal_init_inlock (void **list, const CK_VERSION *version)
```

Comment:
The naming sounds good. The cast can also work, but it will need several casts back and forth to make all the compilers happy. From there, I though using void pointers will save us some of the casting, but if you prefer, I can move to the 3_0 structs.